### ISSUE TITLE
Review if `aria-label` in header navigation is necessary

### ISSUE THREAD

**Issue by hannalaakso** [alphagov/govuk-frontend#1340]
The [header with navigation](https://design-system.service.gov.uk/components/header/with-service-name-and-navigation/index.html) throws the following warning in https://validator.w3.org/:
> Possible misuse of aria-label.

This relates to https://github.com/alphagov/govuk-frontend/blob/fea29aaa833d5f8233c71e0d076923a697e752cd/src/components/header/template.njk#L68 which was inherited from [govuk_template](https://alphagov.github.io/govuk_template/example-proposition-menu.html).

The [WCAG specs ](https://www.w3.org/TR/WCAG20-TECHS/ARIA6.html#ARIA6-examples) state that `aria-label` can be used for:
- ` Distinguishing navigation landmarks`
- `Identifying region landmarks`

We should review whether having `aria-label` in this context provides useful information for assistive technologies. 

There is also a related issue where we hide the list inside the navigation element when showing/hiding the navigation. This means that a user can end up with an empty `<nav></nav>` element, which may cause issues/confusion for screenreader users. See [comment](https://github.com/alphagov/govuk-frontend/issues/1340#issuecomment-501318558) for more context.

## Done when

- [ ] TBC
- [ ] "Details of breaking change" section below completed to help with writing release notes

## Details of breaking change

- which users are affected: TBC
- the change that’s been made: TBC
- changes users will have to make: TBC
- impact of users not making those changes: TBC

**Comment by 36degrees** [comment]
I wonder if the label should be applied to the `nav`, rather than the `ul`.

**Comment by hannalaakso** [comment]
Yes that seems like a potential fix 👍 Tested moving it to `nav` and it's no longer flagged as an error. It also makes sense about what W3C says about
> Distinguishing navigation landmarks

**Comment by 36degrees** [comment]
Tried an initial stab at this in #1440 but it needs more investigation.

Specifically:

- whether the relationship between the `aria-label` on the `ul` and the button which controls the `ul` (rather than the `nav`) is important
- whether it's weird that the `ul` rather than the `nav` collapses, which means for AT users there will always be a `nav` landmark which _may_ be empty.

**Comment by hannalaakso** [comment]
Notes from team discussion:
- It can be useful to be able to label landmarks if there are multiple instances of the same thing on the page and you want to differentiate between them

**Comment by EoinShaughnessy** [comment]
Just want to check my understanding...Is the breaking change here that we're going to ask users to apply `aria-label` to `nav` instead of `ul`? Also, what happens if they do not do this following the release?

**Comment by vanitabarrett** [comment]
@EoinShaughnessy From my understanding at the moment, yes, that may be the change that users need to apply. I think it's definitely still possible that solution may change as I think it still needs testing with different assistive technologies to check how it performs.

There's also the related issue that because of the way the navigation is implemented, you can end up with empty `<nav></nav>` tags with no navigation inside - this is something that may confuse screenreader users. If we decide to make changes to fix that issue too, then there may be more changes for users to make.

**Comment by EoinShaughnessy** [comment]
Thanks, @vanitabarrett!
